hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -619,6 +644,11 @@ func (r *HostedClusterReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 			log.Info("Hosted Cluster is not supported by operator configuration, reconciliation is blocked", "message", supportedHostedCluster.Message)
 			return ctrl.Result{}, nil
 		}
+		validReleaseImage := meta.FindStatusCondition(hcluster.Status.Conditions, string(hyperv1.ValidReleaseImage))
+		if validReleaseImage != nil && validReleaseImage.Status == metav1.ConditionFalse {
+			log.Info("Release image is invalid, reconciliation is blocked", "message", validReleaseImage.Message)
+			return ctrl.Result{}, nil
+		}
```

Comment:
Is the `ValidReleaseImage` condition the only condition which should contribute to a HostedCluster becoming implicitly paused? For example, what about the existing `ValidConfiguration`?

Also, consider we already have a condition `ReconciliationPaused` which causes reconciliation to be paused based on the explicit `.spec.pausedUntil` field. It seems a little awkward to have some arbitrary condition (`ValidReleaseImage`) cause an implicit pause, especially since we don't surface anything to the user indicating the pause in this proposal.

This also makes me wonder if there's some relationship with the [non-upgradeable cluster](https://issues.redhat.com/browse/HOSTEDCP-390) proposal. For example, might `ValidReleaseImage=False` and/or `ValidConfiguration=False` conditions be additional inputs to a higher level `Upgradeable=False` condition, where `Upgradeable` would be the general condition that would cause reconciliation to become implicitly paused (with an aggregation of reasons why being indicated, e.g. "release image is invalid, config is invalid, CVO indicates non-upgradeable, ...")?

Reply:
`ValidConfiguration` already blocks a few lines avoid this.

Yes, there are higher level things that we need to solve outside the scope of this PR.  Namely:

- Reporting to to the user that HC reconciliation/upgrades are blocked on a condition
- Including the status of the underlying CVO upgradeable and ClusterOperator upgradeable conditions